File: algoritmi/esercizi2023/settimana1/istogramma_a_anagrammi/esercizio_3/anagrammi.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	var word1 string
	var word2 string
	var err error

	fmt.Println("Inserisci una parola.")
	_, err = fmt.Scan(&word1)
	if err != nil {
		fmt.Println("Si è verificato un errore nell'input")
		os.Exit(-1)
	}

	fmt.Println("Inserisci una seconda frase.")
	_, err = fmt.Scan(&word2)
	if err != nil {
		fmt.Println("Si è verificato un errore nell'input")
		os.Exit(-1)
	}

	if len(word1) == len(word2) {
		for _, r := range word1 {
			if !strings.Contains(word2, string(r)) {
				fmt.Println("Le due parole non sono l'una l'anagramma dell'altra.")
				os.Exit(0)
			}
		}
		fmt.Println(word2, " è l'anagramma di ", word1)
	} else {
		fmt.Println("Le due parole non sono l'una l'anagramma dell'altra.")
	}
}

File: algoritmi/exercises2021/2021-10-13/anagrammi/anagrammi.c
#include <stdio.h>
#include <ctype.h>

int main() {
    char c,  str1[26] = {0}, str2[26] = {0};
    
    // 4n confronti
    while((c = getchar()) != '\n') {
        if (isalpha(c)) {
            c = tolower(c);
            str1[c - 97]++;
        }
    }

    while((c = getchar()) != '\n') {
        if (isalpha(c)) {
            c = tolower(c);
            str2[c - 97]++;
        }
    }
    // ciclo eseguito 26 volte
    for (int i = 0; i < 26; i++) {
        if (str1[i] != str2[i]) {
            printf("Le due parole non sono l'una l'anagramma dell'altra\n");
            return 0;
        }
    }
    printf("Le due parole sono l'una l'anagramma dell'altra\n");
}

